src/backend/distributed/shared_library_init.c

```diff
@@ -331,6 +331,20 @@ RegisterCitusConfigVariables(void)
 		NULL, NULL, NULL);
 
 	DefineCustomBoolVariable(
+		"citus.enable_deadlock_prevention",
+		gettext_noop("Prevents transactions from expanding to multiple nodes"),
+		gettext_noop("When enabled, transactions that attempt to write to multiple "
+					 "shards on different nodes throw an error to prevent such "
+					 "transactions from creating a distributed deadlock when "
+					 "performed concurrently."),
```

Comment:
Hm. Not yet perfectly happy with the phrasing here, but it's actually not easy to come up with a concise description :(. We allow to write to multiple shards in a fair number of cases even with it enabled, and it can involve multiple nodes...  But I'm also not sure that "expanding the set of modified nodes in consecutive statements within a transaction" is going to be easier to understand :(